src/client/api/v1/get_balance.c

```diff
@@ -0,0 +1,170 @@
+#include "client/api/v1/response_error.h"
+#include "client/network/http.h"
+#include "core/utils/iota_str.h"
+
+int deser_balance_info(char const *const j_str, res_balance_t *res) {
+  char const *const key_addr = "address";
+  char const *const key_maxResults = "maxResults";
+  char const *const key_count = "count";
+  char const *const key_balance = "balance";
+  char const *const key_code = "code";
+  char const *const key_message = "message";
+  double *number;
+  int ret = 0;
+
+  cJSON *json_obj = cJSON_Parse(j_str);
+  if (json_obj == NULL) {
+    return -1;
+  }
+
+  cJSON *error_obj = cJSON_GetObjectItemCaseSensitive(json_obj, key_error);
+  if (error_obj) {
+    char *code = calloc(32, sizeof(char));
+
+    // gets err code
+    if ((ret = json_get_string(error_obj, key_code, code, 32)) != 0) {
+      printf("[%s:%d]: gets %s json error code failed\n", __func__, __LINE__, key_addr);
+      ret = -1;
+      free(code);
+      goto end;
+    }
+
+    if (strcmp(code, "invalid_data") == 0) {
+      printf("[%s:%d]: http code 400\n", __func__, __LINE__, key_addr);
+      ret = -1;
+      free(code);
+      goto end;
+
+    } else if (strcmp(code, "not_found") == 0) {
+      printf("[%s:%d]: http code 404\n", __func__, __LINE__, key_addr);
+      ret = -1;
+      free(code);
+      goto end;
+    }
+
+    free(code);
+  }
+
+  cJSON *data_obj = cJSON_GetObjectItemCaseSensitive(json_obj, key_data);
+  if (data_obj) {
+    // gets addr
+    char *addr = calloc(64, sizeof(char));
+    if ((ret = json_get_string(data_obj, key_addr, addr, 64)) != 0) {
+      printf("[%s:%d]: gets %s json addr failed\n", __func__, __LINE__, key_addr);
+      ret = -1;
+      goto end;
+    }
+
+    if (strlen(addr) == 64) {
+      hex2bin(addr, res->addr + 1, IOTA_ADDRESS_BYTES - 1);
+      res->addr[0] = ADDRESS_VER_ED25519;  // Ed25519
+    } else {
+      printf("[%s:%d]: gets %s json addr failed\n", __func__, __LINE__, key_addr);
+      ret = -1;
+      goto end;
+    }
+
+    free(addr);
+
+    // gets max_results
+    if ((ret = json_get_uint16(data_obj, key_maxResults, res->max_results)) != 0) {
+      printf("[%s:%d]: gets %s json max_results failed\n", __func__, __LINE__, key_maxResults);
+      ret = -1;
+      goto end;
+    }
+
+    res->max_results = (uint16_t)*number;
+
+    // gets count
+    if ((ret = json_get_uint16(data_obj, key_count, res->count)) != 0) {
+      printf("[%s:%d]: gets %s json count failed\n", __func__, __LINE__, key_count);
+      ret = -1;
+      goto end;
+    }
+
+    res->count = (uint16_t)*number;
+
+    // gets balance
+    if ((ret = json_get_uint64(data_obj, key_balance, res->balance)) != 0) {
+      printf("[%s:%d]: gets %s json balance failed\n", __func__, __LINE__, key_balance);
+      ret = -1;
+      goto end;
+    }
+
+    res->balance = (int64_t)*number;
+
+  end:
+    cJSON_Delete(json_obj);
+    return ret;
+  }
+}
+
+int get_balance(iota_client_conf_t const *conf, char *addr, res_balance_t *res) {
```

Comment:
this parameter is an array not just a pointer, for readability I prefer
```suggestion
int get_balance(iota_client_conf_t const *conf, char addr[], res_balance_t *res) {
```